src/server/pfs/server/master.go

```diff
@@ -139,6 +157,119 @@ func (d *driver) finishCommits(ctx context.Context) (retErr error) {
 	)
 }
 
+func (d *driver) manageBranches(ctx context.Context, repoKey string) error {
+	ctx, cancel := pctx.WithCancel(ctx)
+	defer cancel()
+	type cronTrigger struct {
+		cancel context.CancelFunc
+		spec   string
+	}
+	cronTriggers := make(map[string]*cronTrigger)
+	defer func() {
+		for _, ct := range cronTriggers {
+			ct.cancel()
+		}
+	}()
+	return d.branches.ReadOnly(ctx).WatchByIndexF(pfsdb.BranchesRepoIndex, repoKey, func(ev *watch.Event) error {
+		if ev.Type == watch.EventError {
+			return ev.Err
+		}
+		key := string(ev.Key)
+		if ev.Type == watch.EventDelete {
+			if ct, ok := cronTriggers[key]; ok {
+				ct.cancel()
+			}
+			delete(cronTriggers, key)
+			return nil
+		}
+		branchInfo, err := d.inspectBranch(ctx, pfsdb.ParseBranch(key))
+		if err != nil {
+			return err
+		}
+		if branchInfo.Trigger == nil || branchInfo.Trigger.CronSpec == "" {
```

Comment:
shouldn't we also `delete(cronTriggers, key)` here? If a user updates their trigger from `!= nil -> == nil` we would be keeping the goro alive

Reply:
ahh, good catch. Yes, I will change that.